### PR TITLE
Add detection of p4-14 pvs

### PR DESCRIPTION
**Pull request by hesingh**
See 

https://github.com/p4lang/p4c/pull/2183#discussion_r374828806

https://github.com/p4lang/behavioral-model/issues/854

and

https://github.com/p4lang/p4c/issues/2188

With this detection of p4-14, the json output is now different between p4-16 and p4-14.  Now, I am waiting for the discussion to resolve for how is p4-14 mask stored in IR so that the bmv2 backend can emit appropriate JSON.

### CODE REVIEW

**Comment by hesingh** [comment]
Regarding https://github.com/p4lang/p4c/issues/2188,

If one sees the code changes in this PR, viz., this change:

https://github.com/p4lang/p4c/pull/2194/files#diff-2bb7830fa23705a8b613d9b7a96db48fR486

The P4ValueSet constructor has no mask field.   At this juncture where the constructor is called the p4-14 code has been translated to p4-16 for pvs.  Clearly, no mask from p4-14 is carried over to p4-16. So how can bmv2 JSON cater to the p4-14 mask?  The bmv2 backend cannot.

**Comment by jafingerhut** [comment]
One possibility, and we will not know until questions around issue https://github.com/p4lang/p4c/issues/2188 have been answered, discussed, argued over, picked apart in detail, etc., is that Vladimir or someone will come up with a way to translate each of the P4_14 examples into an equivalent P4_16 program that does not use the `pvs mask 0xff00` syntax, or anything like it, but instead perhaps introduces formulas in the select expression instead.

I do not know the intended behavior that Vladimir or others have in their heads for all of those code snippets I asked about.  Until I know the intended meaning, I don't know if that is actually possible or not.

In short: I would suggest not worrying about the details of how to modify p4c, until we actually know how this language feature works.  Without that knowledge, changes to p4c are premature.

**Comment by hesingh** [comment]
Right.  I am not modifying any more p4c.  This changes in the PR just show how to differentiate between p4-14 and p4-16 pvs and now the output JSON is not the same for each type of language program test.   I have no plans to merge this PR yet.

It's only looking at code that I asked my question.  It's a pity that p4_14_samples does not include a test using mask.

No worries.  After this note, I will wait for Vladimir or anyone else.

**Comment by hesingh** [comment]
I am going by what Antonin gave for guidance in https://github.com/p4lang/p4c/pull/2183
He said "To support P4_14 masks, you would probably need some code to distinguish between missing mask and mask set to 0."
